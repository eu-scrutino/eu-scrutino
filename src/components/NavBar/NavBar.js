import React, { useContext } from 'react'
import AppBar from '@material-ui/core/AppBar'
import clsx from 'clsx'
import Toolbar from '@material-ui/core/Toolbar'
import IconButton from '@material-ui/core/IconButton'
import MenuIcon from '@material-ui/icons/Menu'
import Typography from '@material-ui/core/Typography'
import useStyles from './styles'
import config from '../../config'
import { AppContext } from '../../App'

const NavBar = () => {
    const { open, setOpen } = useContext(AppContext)
    const classes = useStyles()

    return (
        <AppBar
            position="fixed"
            className={clsx(classes.appBar, {
                [classes.appBarShift]: open,
            })}
        >
            <Toolbar>
                <IconButton
                    color="inherit"
                    aria-label="open drawer"
                    onClick={() => setOpen(true)}
                    edge="start"
                    className={clsx(classes.menuButton, open && classes.hide)}
                >
                    <MenuIcon data-testid="menu-button" />
                </IconButton>
                <Typography variant="h6" noWrap>
                    {config.siteData.title}
                </Typography>
            </Toolbar>
        </AppBar>
    )
}

export default NavBar
