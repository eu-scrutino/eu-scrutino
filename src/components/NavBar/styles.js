import { makeStyles } from '@material-ui/core/styles'
import { ui } from '../../constants'

const useStyles = makeStyles((theme) => ({
    appBar: {
        transition: theme.transitions.create(['margin', 'width'], {
            easing: theme.transitions.easing.sharp,
            duration: theme.transitions.duration.leavingScreen,
        }),
    },
    appBarShift: {
        width: `calc(100% - ${ui.DRAWER_WIDTH}px)`,
        marginLeft: ui.DRAWER_WIDTH,
        transition: theme.transitions.create(['margin', 'width'], {
            easing: theme.transitions.easing.easeOut,
            duration: theme.transitions.duration.enteringScreen,
        }),
    },
    menuButton: {
        marginRight: theme.spacing(2),
    },
    hide: {
        display: 'none',
    },
}))

export default useStyles
