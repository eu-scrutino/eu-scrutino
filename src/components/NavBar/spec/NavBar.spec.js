import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import NavBar from '../NavBar'

describe('<Navbar />', () => {
    it('renders correctly', () => {
        const props = { open: false, handleDrawerOpen: () => {} }
        const { container } = render(<NavBar {...props} />)

        expect(container).toMatchSnapshot()
    })

    describe('when the hamburger button is clicked', () => {
        it('', () => {
            const handleDrawerOpenMock = jest.fn()
            const props = {
                open: false,
                handleDrawerOpen: handleDrawerOpenMock,
            }
            const { getByRole } = render(<NavBar {...props} />)
            const hamburgerMenuButton = getByRole('button', {
                name: 'open drawer',
            })

            fireEvent.click(hamburgerMenuButton)

            expect(handleDrawerOpenMock).toHaveBeenCalled()
        })
    })
})
