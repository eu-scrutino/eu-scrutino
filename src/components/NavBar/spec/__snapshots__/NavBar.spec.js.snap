// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-1 mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-3 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            data-testid="menu-button"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
      >
        EU | scrutino
      </h6>
    </div>
  </header>
</div>
`;
